connmgr/connmanager.go

```diff
@@ -288,7 +292,30 @@ func (cm *ConnManager) NewConnReq() {
 		cm.requests <- handleFailed{c, err}
 		return
 	}
-	c.Addr = addr
+
+	host, strPort, err := net.SplitHostPort(addr)
```

Comment:
Great suggestion! Modifying `GetNewAddress` to return a `net.Addr` directly allows for a good bit of code consolidation and allows this PR to keep the bulk of the parsing/lookup logic in a single area. 

I've just pushed a commit which makes the modification you mentioned in your comment above. 

> I expect addrmgr.GetAddress itself should probably also be modified at some point to return net.Addr too, but that can be saved for another PR.

One way to implement this would be to make `wire.NetAddress` adhere to the `net.Addr` interface by adding a `.Network()` method. This would allow the addresses returned by `addrmgr.GetAddress` to be directly passed into this context. 

As an aside, I think the `addrmgr` packaged is currently one of the most neglected packages within the project. The code could use a bit of cleaning up, and we could also dump the json file dump format in favor of creating a dedicated bucket for the data within the database. Additionally, it'd be nice to implement some of the counter measures for the attacks described [here](https://www.usenix.org/system/files/conference/usenixsecurity15/sec15-paper-heilman.pdf) and [here](https://arxiv.org/abs/1605.07524).